src/Sylius/Bundle/ApiBundle/Doctrine/QueryItemExtension/OrderMethodsItemExtension.php

```diff
@@ -57,15 +57,20 @@ public function applyToItem(
         $rootAlias = $queryBuilder->getRootAliases()[0];
         $user = $this->userContext->getUser();
 
-        $this->applyUserRulesToItem($user, $queryBuilder, $rootAlias, $httpRequestMethodType);
+        $this->applyUserRulesToItem($user, $queryBuilder, $rootAlias, $httpRequestMethodType, $operationName);
     }
 
     private function applyUserRulesToItem(
         ?UserInterface $user,
         QueryBuilder $queryBuilder,
         string $rootAlias,
-        string $httpRequestMethodType
+        string $httpRequestMethodType,
+        string $operationName
     ): void {
+        if ($operationName === 'shop_select_payment_method') {
```

Comment:
Why you disabled this method in all cases? If it is disabled should be removed because this statement block it in all cases.

Reply:
sorry, but I am not sure what do you mean. my intention was return order always if `shop_select_payment_method` was called. What do you mean by "disabled method in all cases"?

Reply:
But after adding this `if`, the order will be returned even if the order doesn't belong to the current user